core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestUnaliasSymbolReferences.java

```diff
@@ -0,0 +1,152 @@
+                (p, session, metadata) -> {
+                    ColumnHandle column = new TpchColumnHandle("nationkey", BIGINT);
+                    Symbol buildColumnSymbol = p.symbol("nationkey");
+                    Symbol buildAlias1 = p.symbol("buildAlias1");
+                    Symbol buildAlias2 = p.symbol("buildAlias2");
+                    Symbol probeColumn1 = p.symbol("s_nationkey");
+                    Symbol probeColumn2 = p.symbol("s_suppkey");
+                    DynamicFilterId dynamicFilterId1 = new DynamicFilterId("df1");
+                    DynamicFilterId dynamicFilterId2 = new DynamicFilterId("df2");
+
+                    return p.join(
+                            INNER,
+                            p.filter(TRUE_LITERAL, // additional filter to test recursive call
+                                    p.filter(
+                                            ExpressionUtils.and(
+                                                    dynamicFilterExpression(metadata, session, probeColumn1, dynamicFilterId1),
+                                                    dynamicFilterExpression(metadata, session, probeColumn2, dynamicFilterId2)),
+                                            p.tableScan(
+                                                    tableHandle(session, probeTable),
+                                                    ImmutableList.of(probeColumn1, probeColumn2),
+                                                    ImmutableMap.of(
+                                                            probeColumn1, new TpchColumnHandle("nationkey", BIGINT),
+                                                            probeColumn2, new TpchColumnHandle("suppkey", BIGINT))))),
+                            p.project(
+                                    Assignments.of(buildAlias1, buildColumnSymbol.toSymbolReference(), buildAlias2, buildColumnSymbol.toSymbolReference()),
+                                    p.tableScan(tableHandle(session, buildTable), ImmutableList.of(buildColumnSymbol), ImmutableMap.of(buildColumnSymbol, column))),
+                            ImmutableList.of(),
+                            ImmutableList.of(),
+                            ImmutableList.of(buildAlias1, buildAlias2),
+                            Optional.empty(),
+                            Optional.empty(),
+                            Optional.empty(),
+                            ImmutableMap.of(dynamicFilterId1, buildAlias1, dynamicFilterId2, buildAlias2));
+                },
+                join(
+                        INNER,
+                        ImmutableList.of(),
+                        ImmutableMap.of("probeColumn1", "column", "probeColumn2", "column"),
+                        filter(TRUE_LITERAL, filter(TRUE_LITERAL, tableScan(probeTable, ImmutableMap.of("probeColumn1", "suppkey", "probeColumn2", "nationkey")))),
+                        project(tableScan(buildTable, ImmutableMap.of("column", "nationkey")))));
```

Comment:
We could have a more straightforward way of testing plans with dynamic filters. This is hacky and misleading.

This is just a side comment, though, not a request for change.

Reply:
Agreed, @sopel39 had similar comments here as well. it's even hard to see where the DF is being tested for all the noise.
I appreciate any ideas on how to improve it (I could follow up on this in another PR)